src/dsp_tools/commands/xmlupload/xmlupload.py

```diff
@@ -397,6 +398,9 @@ def _upload_resources(
     )
 
     total_res = len(resources)
+    # if the interrupt_after value is not set, the upload will not be interrupted
+    interrupt_after = config.interrupt_after or total_res + 1
+
```

Comment:
i think these lines should be at the very beginning of the for loop, i.e. on line 405